justeattakeaway/pie
# feat(wc-next13): Render button via SSR in Next@13

## Describe your changes (can list changeset entries if preferable)

this PR integrates [@lit-labs/nextjs ](https://github.com/lit/lit/tree/main/packages/labs/nextjs) with our next@13 example to ensure _full_ SSR support for our button component. 

**How to test?** 
- open up the page source tab and see the generated markup in the server 🔥 
<img width="1440" alt="Screenshot 2023-05-10 at 12 01 47" src="https://github.com/justeattakeaway/pie/assets/28605803/deb69f6d-1389-4620-9b14-15b133a35cc3">

**Known bugs by the Lit team:** 

1- Running `yarn dev` to start the app locally without appending `NODE_OPTIONS=--conditions=development` will point some of the lit dependencies to production bundles instead of development and that throws run time errors. check [here](https://github.com/lit/lit/issues/3845) for more details 

## Author Checklist (complete before requesting a review)
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

## Reviewer checklists (complete before approving)
### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them
